packages/recordio/src/read.js

```diff
@@ -0,0 +1,22 @@
+const RECORD = /^\d+\n.+/;
+
+export default function read(input, records = []) {
```

Comment:
I like that you took the time to improve this function. Your implementation is now way more readable. Unfortunately the implementation is way slower than your initial one as your unnecessarily executing string operations.

I ran a benchmark using your initial, the current implementation as well as one I'd propose. 

> read (initial) x 1,921,278 ops/sec ±0.69% (87 runs sampled)
> read (current) 794,427 ops/sec ±0.55% (84 runs sampled)
> read (proposed) x 1,907,616 ops/sec ±1.42% (82 runs sampled)

The numbers vary a little but the following my proposal is either as good as imperative one or beats it by a few ops/sec. 

```js
const RECORD_PATTER = /^(\d+)\n([^]*)/

function read(input, records) {
  const [_, length, data] = RECORD_PATTER.exec(input) || [];

  if (!length || !data || data.length < length) {
    return [records, input];
  }

  const record = data.substring(0, length);
  const rest = input.substring(length + 1);

  return read(rest, records.concat(record));
}
```

Reply:
The solution you've provided doesn't pass tests, but I got the idea! Thanks for the input, I'll try to improve the performance

Reply:
I fixed the solution and eager to know what kind of a benchmark did you use?
```javascript
const RECORD_PATTERN = /^(\d+)\n([^]*)/;

export default function read(input, records = []) {
  let [_, length, data] = RECORD_PATTERN.exec(input) || [];
  length = parseInt(length, 10);

  if (!length || !data || data.length < length) {
    return [records, input];
  }

  const record = data.substr(0, length);
  const rest = data.substr(length);

  return read(rest, records.concat(record));
}
```